frontend/src/hooks/useGetHistoryEvents.ts

```diff
@@ -0,0 +1,37 @@
+import { EVENTS_QUERY_KEY } from '@/queryKeyStore'
+import { TokenPreview } from '@/types'
+import { wagmiConfig } from '@/wagmi.config'
+import { useQuery } from '@tanstack/react-query'
+import { getPublicClient } from '@wagmi/core'
+import { sepolia } from '@wagmi/core/chains'
+import { Log } from 'viem'
+
+export const useGetHistoryEvents = (selectedToken: TokenPreview, limit: number) => {
+  const publicClient = getPublicClient(wagmiConfig, { chainId: sepolia.id })
+
+  const { data: latestBlock } = useQuery({
+    queryKey: ['latestBlock', { tokenTitle: selectedToken.title }],
+    queryFn: () => publicClient.getBlockNumber(),
+  })
+
+  const queryData = useQuery<Log[]>({
+    queryKey: [EVENTS_QUERY_KEY, { limit, tokenTitle: selectedToken.title }],
+    queryFn: () => {
+      const fromBlockLimit = latestBlock! - 10000n
```

Comment:
There could be an issue if the contract was deployed before that, instead it would be better to have config in which we would save the deployment block and read all events starting from this block. That would also remove the edge case if the from block is lower than 0